demo/apisix/tmail-apisix-plugin-runner/src/main/java/com/linagora/apisix/plugin/RewriteXUserFilter.java

```diff
@@ -79,7 +79,7 @@ public Optional<String> extractUserFromUserInfo(String userInfo) {
             }
             logger.warn("Wrong userinfo field userInfo {}", jsonNode.getClass());
             return Optional.empty();
-        } catch (IOException e) {
+        } catch (NullPointerException | IOException e) {
```

Comment:
It would be more explicit to check explicitly that `json.readTree(Base64.getDecoder().decode(userInfo))` contains `userInfoField` field and avoid generating the `NullPointerException` altogether IMO ...

(Akka do not rely on error handling when it is not truely needed)